core/src/main/scala/org/apache/spark/ui/JettyUtils.scala

```diff
@@ -208,6 +210,13 @@ private[spark] object JettyUtils extends Logging {
         rewrittenURI.toString()
       }
 
+      override def newHttpClient(): HttpClient = {
+        // Use the Jetty logic to calculate the number of selector threads (#CPUs/2),
+        // but limit it to 8 max.
```

Comment:
Could you also explain why we are limiting the number of selectors in the comment above?